opencog/atomspace/TruthValue.cc

```diff
@@ -94,3 +94,13 @@ bool TruthValue::isDefaultTV() const
     }
     return false;
 }
+
+TruthValueType TruthValue::getType() const
+{
+	return tv_type_;
+}
+
+void TruthValue::setType(TruthValueType t)
+{
+	tv_type_ = t;
```

Comment:
This is deeply and fundmentally wrong.  The different C++ classes have different numbers or members (simpe has 2, count has 3) and different merge methods. The numbers all have different meanings in the different classes. Changing the type is just .. impossible.